src/compiler/scala/tools/nsc/ast/parser/Parsers.scala

```
@@ -2742,20 +2742,17 @@ self =>
      *  TraitParents       ::= AnnotType {with AnnotType}
      *  }}}
      */
-    def templateParents(isTrait: Boolean): (List[Tree], List[List[Tree]]) = {
-      val parents = new ListBuffer[Tree] += startAnnotType()
-      val argss = (
-        // TODO: the insertion of ListOfNil here is where "new Foo" becomes
-        // indistinguishable from "new Foo()".
```

Yes. `typedParentType` and `internal.Trees.New` still silently convert Nil to ListOfNil, but it's their own choice, no longer imposed by the typer. Upd. typo: This should read "by the parser".